timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/TimeLockServerConfiguration.java

```diff
@@ -76,4 +84,27 @@ public ClusterConfiguration cluster() {
     public long slowLockLogTriggerMillis() {
         return 10000;
     }
+
+    public boolean useThreadPooling() {
+        return useThreadPooling;
+    }
+
+    public int availableThreads() {
+        if (!useThreadPooling()) {
+            throw new IllegalStateException("Should not call availableThreads() if useThreadPooling is disabled");
+        }
+
+        return computeNumberOfAvailableThreads();
+    }
+
+    private int computeNumberOfAvailableThreads() {
+        DefaultServerFactory serverFactory = (DefaultServerFactory) getServerFactory();
+        int maxServerThreads = serverFactory.getMaxThreads();
+        HttpConnectorFactory connectorFactory = (HttpConnectorFactory) serverFactory.getApplicationConnectors().get(0);
```

Comment:
Both cases have a simple runtime check we can add:
- `Preconditions.checkState` for the class of getServerFactory()
- `Preconditions.checkNotNull` for the application connectors